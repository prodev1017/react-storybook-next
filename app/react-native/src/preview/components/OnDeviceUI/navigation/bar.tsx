import React, { PureComponent } from 'react';
import styled from '@emotion/native';
import Button from './button';
import { NAVIGATOR, PREVIEW, ADDONS } from './constants';

const Container = styled.View(({ theme }) => ({
  flexDirection: 'row',
  paddingHorizontal: 8,
  backgroundColor: theme.backgroundColor,
  borderTopWidth: 1,
  borderBottomWidth: 1,
  borderColor: theme.borderColor,
}));

export interface Props {
  index: number;
  onPress: (id: number) => void;
}

export default class Bar extends PureComponent<Props> {
  render() {
    const { index, onPress } = this.props;
    return (
      <Container>
        <Button onPress={onPress} id={NAVIGATOR} active={index === NAVIGATOR}>
          NAVIGATOR
        </Button>
        <Button onPress={onPress} id={PREVIEW} active={index === PREVIEW}>
          PREVIEW
        </Button>
        <Button onPress={onPress} id={ADDONS} active={index === ADDONS}>
          ADDONS
        </Button>
      </Container>
    );
  }
}
