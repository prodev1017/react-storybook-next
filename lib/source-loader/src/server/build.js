import { getOptions } from 'loader-utils';
import { readStory } from './dependencies-lookup/readAsObject';
import { getRidOfUselessFilePrefixes } from './dependencies-lookup/getRidOfUselessFilePrefixes';
import getParser from './abstract-syntax-tree/parsers';
import { endOfImports } from './abstract-syntax-tree/traverse-helpers';

export function insertAfterImports(classLoader, insert, source) {
  const options = getOptions(classLoader) || {};
  let ast;
  try {
    ast = getParser(options.parser || classLoader.extension || 'javascript').parse(source);
  } catch (e) {
    // if not working, then we will fallback to not adding anything
    // perhaps the code was not written in javascript
    return source;
  }
  if (!ast) return `${insert}${source}`;
  const endOfImportsIndex = endOfImports(ast);
  const result = `${source.substring(0, endOfImportsIndex)}\n${insert}\n${source.substring(
    endOfImportsIndex
  )}`;
  return result;
}

export function transform(inputSource) {
  return readStory(this, inputSource)
    .then(getRidOfUselessFilePrefixes)
    .then(
      ({
        prefix,
        resource,
        source,
        sourceJson,
        addsMap,
        dependencies,
        localDependencies,
        idsToFrameworks,
      }) => {
        const preamble = `
/* eslint-disable no-unused-vars,@typescript-eslint/no-unused-vars */
// @ts-ignore
var withSourceLoader = require('@storybook/source-loader/preview').withSource;
// @ts-ignore
var addSourceDecorator = require("@storybook/source-loader/preview").addSource;
// @ts-ignore
var __SOURCE_PREFIX__ = "${prefix.replace(/\\([^\\ ])/g, '\\\\$1')}";
// @ts-ignore
var __STORY__ = ${sourceJson};
// @ts-ignore
var __ADDS_MAP__ = ${JSON.stringify(addsMap)};
// @ts-ignore
var __MAIN_FILE_LOCATION__ = ${JSON.stringify(resource)};
// @ts-ignore
var __MODULE_DEPENDENCIES__ = ${JSON.stringify(dependencies)};
// @ts-ignore
var __LOCAL_DEPENDENCIES__ = ${JSON.stringify(localDependencies)};
// @ts-ignore
var __IDS_TO_FRAMEWORKS__ = ${JSON.stringify(idsToFrameworks)};
/* eslint-enable no-unused-vars,@typescript-eslint/no-unused-vars */
        `;
        return insertAfterImports(this, preamble, source);
        // return `${preamble}${source}`;
      }
    );
}
