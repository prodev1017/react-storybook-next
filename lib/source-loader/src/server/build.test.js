import { insertAfterImports } from './build';

const insert = 'INSERT\n';

describe('insertAfterImports', () => {
  it('no imports', () => {
    const noImports = `
foo bar;
baz;
    `.trim();
    expect(insertAfterImports({}, insert, noImports)).toMatchSnapshot();
  });

  it('imports', () => {
    const hasImports = `
import './foo';
import './bar';
whatever;
    `.trim();
    expect(insertAfterImports({}, insert, hasImports)).toMatchSnapshot();
  });

  it('single-line imports', () => {
    const hasImports = `
import 'foo';
import { bar } from 'baz';
whatever;
    `.trim();
    expect(insertAfterImports({}, insert, hasImports)).toMatchSnapshot();
  });

  it('multi-line imports', () => {
    const hasImports = `
import 'foo';
import {
  bar
} from 'baz';
whatever;
    `.trim();
    expect(insertAfterImports({}, insert, hasImports)).toMatchSnapshot();
  });

  it('addon-notes', () => {
    const notesStory = `
    import React from 'react';

import BaseButton from '../components/BaseButton';
import markdownNotes from './notes/notes.md';

const markdownString = '...';
    `.trim();
    expect(insertAfterImports({}, insert, notesStory)).toMatchSnapshot();
  });
});
