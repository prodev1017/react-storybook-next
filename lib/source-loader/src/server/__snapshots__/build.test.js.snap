// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insertAfterImports addon-notes 1`] = `
"import React from 'react';

import BaseButton from '../components/BaseButton';
import markdownNotes from './notes/notes.md';
INSERT



const markdownString = '...';"
`;

exports[`insertAfterImports imports 1`] = `
"import './foo';
import './bar';
INSERT


whatever;"
`;

exports[`insertAfterImports multi-line imports 1`] = `
"import 'foo';
import {
  bar
} from 'baz';
INSERT


whatever;"
`;

exports[`insertAfterImports no imports 1`] = `
"
INSERT

foo bar;
baz;"
`;

exports[`insertAfterImports single-line imports 1`] = `
"import 'foo';
import { bar } from 'baz';
INSERT


whatever;"
`;
